Tests/Delphi.Mocks.Tests.Utils.pas

```diff
@@ -113,6 +150,357 @@ procedure TTestTValue.Test_TValue_NotEquals_Strings;
   CheckFalse(v1.Equals(v2));
 end;
 
+
+procedure TTestTValue.Test_TValue_Equals_Variants;
+var
+  var1,var2 : variant;
+  v1, v2 : TValue;
+begin
+  var1 := 'hello';
+  var2 := 'hello';
+  v1 := TValue.FromVariant( var1 );
+  v2 := TValue.FromVariant( var2 );
+  CheckTrue(v1.Equals(v2));
+end;
+
+procedure TTestTValue.Test_TValue_Equals_Variants_BothEmpty;
+var
+  var1,var2 : variant;
+  v1, v2 : TValue;
+begin
+  var1 := varEmpty;
+  var2 := varEmpty;
+  v1 := TValue.FromVariant( var1 );
+  v2 := TValue.FromVariant( var2 );
+  CheckTrue(v1.Equals(v2));
+end;
+
+procedure TTestTValue.Test_TValue_Equals_Variants_BothUnAssigned;
+var
+  var1,var2 : variant;
+  v1, v2 : TValue;
+begin
+  var1 := UnAssigned;
+  var2 := UnAssigned;
+  v1 := TValue.FromVariant( var1 );
+  v2 := TValue.FromVariant( var2 );
+  CheckTrue(v1.Equals(v2));
+end;
+
+procedure TTestTValue.Test_TValue_Equals_Variants_VarArray;
+var
+  var1,var2 : variant;
+  v1, v2 : TValue;
+begin
+  var1 := VarArrayCreate( [0,3], varVariant );
```

Comment:
What you are passing here are the bounds of the array to be created. So it creates an array with 4(!) elements but you only assign the first 3 items. This might be why you were seeing some wrong results earlier and did the -1 on the VarArrayHighBound. This is defect across all tests for varArray.